presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -193,6 +193,8 @@
 
     private Duration partitionLeaseDuration = new Duration(0, TimeUnit.SECONDS);
 
+    private boolean executionBasedMemoryAccounting;
```

Comment:
should we name this flag, includeDictionaryRowGroupBytes ?

By default set this to false, and add a comment saying that enabling this flag breaks multiple queries to fail with OOM, though this is the right behavior.

Reply:
Writer Configs are maintained in the file https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterConfig.java

HiveClientConfig usually only contains the reader configs.

Reply:
This just the first memory accounting fix we want to be able to turn off for environments that are memory constrained and would see too many failing queries if the bug were fixed.

We could have a flag for each one, but it will be simpler to just have a loose memory accounting flag that documents why some people would prefer loose memory accounting. The long term plan is that we want to be able to adapt to memory pressure so that we can do accurate accounting, but until that is implemented we want to continue to allow queries to use extra memory.

I don't think we want to default to broken memory accounting. Most users are not that memory constrained and most don't have a very large body of scheduled and repeating queries. Those that do are probably more sophisticated and can set this flag based on the release note.

Reply:
Having a single flag that controls all the setting is fine. But we should not propagate the same setting to the OrcWriters. OrcWriters should expose different options and a higher layer can control the mappings. For example, there can be one memoryRelateSetting and it gets transferred to few OrcWriter related settings, but the setting should be specific to OrcWriter. Having a setting that is on a high level to OrcWriter, will be harder to maintain in the Orcwriter.

Reply:
`executionBasedMemoryAccounting` is not an appropriate name for this flag because it has nothing to do with execution modes or Presto on Spark.

This flag exists because a small number of users don't want accurate memory accounting because it breaks a small % of their workload's queries. The reason it is a small number of users is because very few people have enough impacted queries where they would want to trade off accuracy for having their queries continue to run.

I think this should just communicate that it enables inaccurate or loose memory accounting. How exactly is pretty arbitrary because once you are enabling this kind of option it's because you care more about not breaking queries than having accurate accounting.